tests/unit/test_process_functions.py

```diff
@@ -0,0 +1,151 @@
+import pandas as pd
+import genie
+import pytest
+
+DATABASE_DF = pd.DataFrame({
+    'UNIQUE_KEY': ['test1', 'test2', 'test3'],
+    "test": ['test1', 'test2', 'test3'],
+    "foo": [1, 2, 3],
+    "baz": [float('nan'), float('nan'), float('nan')]})
+DATABASE_DF.index = ['1_3', '2_3', '3_5']
+
+
+def test_valid__check_valid_df():
+    genie.process_functions._check_valid_df(DATABASE_DF, "test")
+
+
+def test_invalid__check_valid_df():
+    with pytest.raises(ValueError):
+        genie.process_functions._check_valid_df("foo", "test")
+    with pytest.raises(ValueError):
+        genie.process_functions._check_valid_df(DATABASE_DF, "error")
+
+
+def test__get_left_diff_df():
+    new_datadf = pd.DataFrame({
+        'UNIQUE_KEY': ['test1', 'test2', 'test3', 'test4'],
+        "test": ['test1', 'test2', 'test3', 'test4'],
+        "foo": [1, 2, 3, 4],
+        "baz": [float('nan'), float('nan'), float('nan'), 3.2]})
+    get_diff = genie.process_functions._get_left_diff_df(
+        new_datadf, DATABASE_DF, 'UNIQUE_KEY')
+    expecteddf = new_datadf.loc[[3]]
+    assert get_diff.equals(expecteddf[get_diff.columns])
+
+
+def test_norows_get_left_diff_df():
```

Comment:
Add a test case for invalid inputs -- only need to check that validation is performed.